Common/Interfaces/ITimeInForceHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Orders;
+
+namespace QuantConnect.Interfaces
+{
+    /// <summary>
+    /// Handles the time in force for an order
+    /// </summary>
+    public interface ITimeInForceHandler
+    {
+        /// <summary>
+        /// Checks if an order has expired
+        /// </summary>
+        /// <param name="order">The order to be checked</param>
+        /// <returns>Returns true if the order has expired, false otherwise</returns>
+        bool HasOrderExpired(Order order);
```

Comment:
We spoke about adding the `Security` instance here to match existing patterns w.r.t models in lean. For example, look at the fee/fill/slippage models, all of which accept the security object in addition to the order object. This will remove the need for the algorithm and market hours db dependencies